packages/mcp/src/server.ts

```diff
@@ -111,16 +182,28 @@ export class MCPServer extends MCPServerBase {
 
       const coreTool = makeCoreTool(toolInstance, options) as InternalCoreTool;
 
-      convertedTools[toolName] = {
+      explicitConvertedTools[toolName] = {
         name: toolName,
         description: coreTool.description,
         parameters: coreTool.parameters,
         execute: coreTool.execute!,
       };
-      this.logger.info(`Registered tool: '${toolName}' [${toolInstance?.description || 'No description'}]`);
+      processedToolNames.add(toolName);
+      this.logger.info(`Registered explicit tool: '${toolName}'`);
     }
-    this.logger.info(`Total tools registered: ${Object.keys(convertedTools).length}`);
-    return convertedTools;
+    this.logger.info(`Total explicit tools registered: ${Object.keys(explicitConvertedTools).length}`);
+
+    // Process agents and convert them to tools
+    const agentDerivedTools = this._convertAgentsToTools(agentsConfig, processedToolNames);
```

Comment:
Not a big deal but if you move the looping logic out to this scope, remove `_convertAgentsToTools` and just have a singular `private agentToTool` function you can simplify the code with less nesting+indirection and you don't need `processedToolNames`, you can just check `if (definedConvertedTools[toolName]) {`